### PR TITLE
fix: cursor is left behind

### PR DESCRIPTION
**Pull request by kuator**
https://stackoverflow.com/questions/49997089/vscode-how-to-set-that-when-i-press-ctrl-up-or-down-arrow-the-cursor-move-wi
Vs Code, by default, doesn't update cursor position after scrolling, which is not the case in vim. This pr makes experience more similar to scrolling in vim.

### CODE REVIEW

**Comment by asvetliakov** [comment]
probably need a setting here

**Comment by kuator** [comment]
Kind of like this?

File: src/commands_controller.ts [hunk 1]
```diff
@@ -108,7 +109,9 @@ export class CommandsController implements Disposable, NeovimExtensionRequestPro
     };
 
     private scrollLine = (to: "up" | "down"): void => {
-        vscode.commands.executeCommand("editorScroll", { to, by: "line", revealCursor: false });
+        const settings = vscode.workspace.getConfiguration(EXT_NAME);
```
Comment (asvetliakov) [inline src/commands_controller.ts:112]: Pass the option as constructor param from the main_controller, don't query it every time for perf reasons

**Review by asvetliakov — CHANGES_REQUESTED** [review]

**Comment by kuator** [comment]
Is it better?

File: src/main_controller.ts [hunk 2]
```diff
@@ -173,7 +174,22 @@ export class MainController implements vscode.Disposable {
         const channel = await this.client.channelId;
         await this.client.setVar("vscode_channel", channel);
 
-        this.commandsController = new CommandsController(this.client);
+        this.logger.debug(`${LOG_PREFIX}: UIAttach`);
+        await this.client.uiAttach(this.NEOVIM_WIN_WIDTH, this.NEOVIM_WIN_HEIGHT, {
```
Comment (asvetliakov) [inline src/main_controller.ts:178]: Broken merge ? remove

**Review by asvetliakov — CHANGES_REQUESTED** [review]
Almost 😄

**Review by asvetliakov — APPROVED** [review]

**Comment by asvetliakov** [comment]
Fixes #236

**Comment by asvetliakov** [comment]
Thank you for your PR!

**Comment by d4h0** [comment]
@kuator: Thanks a million! This problem was one of the main reason why I stopped using this extension in the past.

@asvetliakov: Is there a reason why this can not be activated by default?

And a million thanks to you as well for creating the extension – without the now fixed cursor problem, it's fantastic!

**Comment by kuator** [comment]
> @kuator: Thanks a million! This problem was one of the main reason why I stopped using this extension in the past.
> 
> @asvetliakov: Is there a reason why this can not be activated by default?
> 
> And a million thanks to you as well for creating the extension – without the now fixed cursor problem, it's fantastic!

😢thanks

**Pull request merged**